design-documents/service-isolation/inter-service-communication.md

```diff
@@ -0,0 +1,294 @@
+### Inter-service communications
+This documents describes how exactly services will be communicating in a distributed setup.
+#### Service declaration
+Services will be declared in Api modules using interfaces. These modules will be installed on all nodes.
+Example:
+```php
+namespace Magento\ServiceBApi;
+
+interface BManagerInterface
+{
+    public function save(BDataInterface $data): BDataInterface;
+    
+    public function delete(string $id): void;
+}
+```
+or for DTOs
+```php
+namespace Magento\ServiceBApi\Data;
+
+interface BDataInterface
+{
+    public function getId(): string;
+  
+    public function getName(): string;
+}
+```
+The API modules will just contain these declarations and nothing else (no di.xml, schema.xml, webapi.xml etc).
+ 
+Even services that are not written using Magento (or even PHP) will be declared in this way for nodes using Magento.
+ 
+#### Publicly available Web APIs
+To enable publicly available APIs (that frontend and 3rd party systems can use) the _front node_ will have to have -Module-Webapi
```

Comment:
BFF will expose aggregated operations and operations that needs to be proxied to a service. As I understand this point is about operations that need to be proxied? Maybe we can solve this with configuration?